src/sage/rings/number_field/number_field.py

```diff
@@ -4202,16 +4202,39 @@ def _pari_absolute_structure(self):
             sage: K.<a> = NumberField(2*x^2 + 1/3)
             sage: K._pari_absolute_structure()
             (y^2 + 6, Mod(1/6*y, y^2 + 6), Mod(6*y, y^2 + 1/6))
-        """
-        f = self.absolute_polynomial()._pari_with_name('y')
-        f = f * f.content().denominator()
-        if f.pollead() == 1:
-            g = f
-            alpha = beta = g.variable().Mod(g)
-        else:
-            g, alpha = f.polredbest(flag=1)
-            beta = alpha.modreverse()
-        return g, alpha, beta
+
+        TESTS:
+
+        Checking that the representation in not improved in a costly manner (see :issue:`39920`)::
+
+            sage: from cysignals.alarm import alarm
+            sage: K = NumberField(ZZ['x']([1]*200 + [2]), 'a')
+            sage: QQasNF = NumberField(ZZ['x']([1,-1]), 'b')
+            sage: alarm(0.5) # ensuring that a trivial isomorphism finishes in reasonable time
+            sage: K.is_isomorphic(QQasNF)
+            False
+            sage: cancel_alarm()
+        """
+        g = self.absolute_polynomial()
+        # make it integral
+        g *= g.denominator()
+        scalar = g.leading_coefficient()
+        if scalar != 1:
+            # doing g = g(x/scalar) in linear time
+            from operator import mul
+            from itertools import accumulate
+            # scalar^i
+            powers = accumulate([1/scalar] + [scalar] * g.degree(), mul)
+            # need to double reverse
+            g = g.parent()([c*p for c, p in zip(g.reverse(), powers)]).reverse()
+        g /= gcd(g.coefficients())
```

Comment:
does `g.content()` works here? i.e. is it guaranteed to be an element of ℤ[x]?

If it is mathematically guaranteed but not in the implementation it may be a good idea to `change_ring` it to ZZ. the implement in `.content` is expected to be slightly more efficient.